shared/rakefiles/xk_util.rake

```diff
@@ -55,6 +55,34 @@ task :rotate_secrets_key, [:encryption_key] => [:configure, :configure_secrets]
   @secrets.disable_non_primary_key_versions(args[:encryption_key], new_version_id)
 end
 
+# This task gracefully rotates SA credentials used by target K8s deployment
+task :rotate_deployment_credentials, [:deployment, :namespace] => [:configure] do |taskname, args|
+  if args[:deployment].nil?
+    puts "  ERROR: args[:deployment] must be set!"
+    raise
+  end
+  puts args[:namespace].inspect
+  namespace = args[:namespace] ? args[:namespace] : "gpii"
+
+  # Rolling-restarting deployment so all pods can obtain tokens from static credentials
+  sh "kubectl -n #{namespace} rollout restart deployment #{args[:deployment]}"
+
+  # Rollout status ensures that all pods updated (waiting)
+  sh "kubectl -n #{namespace} rollout status deployment #{args[:deployment]}"
+
+  # Destroying existing credentials
+  sh "rake destroy_sa_keys['gke-cluster-pod-#{args[:deployment]}']"
+
+  # Applying deployment module to force TF to generate new credentials and re-create K8s secret
+  sh "rake xk['apply live/#{@env}/k8s/#{namespace}/#{args[:deployment]}',true,false,true]"
```

Comment:
This is just simplest tool for the purpose I can come up with. We may later replace this with `gcloud` command that generates the new keys and then imports updated resource into TF state, to not "modify the state of the world".